expression/builtin_arithmetic_vec.go

```diff
@@ -429,13 +429,146 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
 }
 
 func (b *builtinArithmeticIntDivideIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinArithmeticIntDivideIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
-}
+	n := input.NumRows()
+	lh, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(lh)
+
+	if err := b.args[0].VecEvalInt(b.ctx, input, lh); err != nil {
+		return err
+	}
+
+	if err := b.args[1].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	rh := result
+	lhi64s := lh.Int64s()
+	rhi64s := rh.Int64s()
+
+	resulti64s := result.Int64s()
+
+	isLHSUnsigned := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	isRHSUnsigned := mysql.HasUnsignedFlag(b.args[1].GetType().Flag)
 
+	switch {
+	case isLHSUnsigned && isRHSUnsigned:
+		err = b.divideUU(result, lh, rh, lhi64s, rhi64s, resulti64s)
+	case isLHSUnsigned && !isRHSUnsigned:
+		err = b.divideUI(result, lh, rh, lhi64s, rhi64s, resulti64s)
+	case !isLHSUnsigned && isRHSUnsigned:
+		err = b.divideIU(result, lh, rh, lhi64s, rhi64s, resulti64s)
+	case !isLHSUnsigned && !isRHSUnsigned:
+		err = b.divideII(result, lh, rh, lhi64s, rhi64s, resulti64s)
+	}
+	return err
+}
+func (b *builtinArithmeticIntDivideIntSig) divideUU(result, lhs, rhs *chunk.Column, lhi64s, rhi64s, resulti64s []int64) error {
+	for i := 0; i < len(lhi64s); i++ {
+		if rhs.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
```

Comment:
After calling `result.MergeNulls(ls)`, we can skip calling `SetNull(i, true)` here.